android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListViewProxy.java

```diff
@@ -375,6 +379,69 @@ public void setActivity(Activity activity)
 		}
 	}
 
+	// NOTE: For internal use only.
+	public KrollDict getContentOffset()
+	{
+		final TiListView listView = getListView();
+
+		if (listView != null) {
+			final KrollDict contentOffset = new KrollDict();
+
+			final int x = (int) new TiDimension(listView.getScrollOffsetX(),
+				TiDimension.TYPE_WIDTH, COMPLEX_UNIT_DIP).getAsDefault(listView);
+			final int y = (int) new TiDimension(listView.getScrollOffsetY(),
+				TiDimension.TYPE_HEIGHT, COMPLEX_UNIT_DIP).getAsDefault(listView);
+
+			contentOffset.put(TiC.PROPERTY_X, x);
+			contentOffset.put(TiC.PROPERTY_Y, y);
+
+			// NOTE: Since obtaining the scroll offset from RecyclerView is unreliable
+			// when items are added/removed, also grab the current visible item instead.
+			final int currentIndex = listView.getAdapterIndex(listView.getFirstVisibleItem().index);
+			contentOffset.put(TiC.PROPERTY_INDEX, currentIndex);
+
+			this.contentOffset = contentOffset;
+		}
+
+		return this.contentOffset;
+	}
+
+	@Kroll.method
+	public void setContentOffset(KrollDict contentOffset, @Kroll.argument(optional = true) KrollDict options)
+	{
+		final TiListView listView = getListView();
+
+		if (contentOffset != null) {
+			this.contentOffset = contentOffset;
+
+			if (listView != null) {
+
+				if (contentOffset.containsKeyAndNotNull(TiC.PROPERTY_INDEX)) {
+
+					// If available, scroll to provided index provided by internal `getContentOffset()` method.
+					listView.getRecyclerView().scrollToPosition(contentOffset.getInt(TiC.PROPERTY_INDEX));
```

Comment:
There is a bug here. This code always assumes the index belongs to the 1st section. _(The `getContentOffset()` does not offset the stored index with the total number of section items above it.)_

In your test code, scroll to the very bottom of the ListView. Note that the top-most visible row should be in `Section B`. Now switch between dark/light mode (or enable/disable editing) and you'll notice that the ListView scrolls to `Section A` equivalent of that row item number.